torch/csrc/generic/Tensor.cpp

```diff
@@ -95,18 +95,14 @@ static void THPTensor_(setInconsistentDepthError)(std::vector<size_t> &sizes,
 #ifdef NUMPY_TYPE_ENUM
 THTensor* THPTensor_(fromNumpy)(PyObject *numpy_array) {
   PyArrayObject *array = (PyArrayObject*)numpy_array;
-  THStoragePtr storage = THStorage_(newWithDataAndAllocator)(
-      (real*)PyArray_DATA(array),
-      PyArray_NBYTES(array) / sizeof(real),
-      &THNumpyArrayAllocator,
-      new NumpyArrayAllocator(numpy_array));
 
   // Numpy and Torch disagree on empty tensors. In Torch, an empty
   // tensor is a tensor with zero dimensions. In Numpy, an empty tensor
   // keeps its shape, but has 0 as the size of one of the dimensions.
   // So we'll convert all Numpy tensors of 0 elements to empty Torch tensors.
   if (PyArray_SIZE(array) != 0) {
     auto ndim = PyArray_NDIM(array);
+    size_t storage_size = 1;
```

Comment:
OK, @apaszke explained this to me. strides * (sizes-1) is the offset of the last element. The storage size has to be the largest offset + 1.